sw/host/opentitanlib/src/transport/hyperdebug/spi.rs

```diff
@@ -0,0 +1,213 @@
+            0 /* wValue */,
+            hyperdebug.spi_interface.interface as u16,
+            &mut [])?;
+
+        // Exclusively claim SPI interface, preparing for bulk transfers.
+        usb_handle.claim_interface(hyperdebug.spi_interface.interface)?;
+
+        // Initial bulk request/response to query capabilities.
+        let mut buf = [0u8; USB_MAX_SIZE];
+        buf[0..2].clone_from_slice(&USB_SPI_PKT_ID_CMD_GET_USB_SPI_CONFIG.to_le_bytes());
+        usb_handle.write_bulk(hyperdebug.spi_interface.out_endpoint, &buf[0..2])?;
+        let rc = usb_handle.read_bulk(hyperdebug.spi_interface.in_endpoint, &mut buf)?;
+        ensure!(rc == 8, Error::CommunicationError("Unrecognized reponse to GET_USB_SPI_CONFIG"));
+        ensure!(buf[0..2] == USB_SPI_PKT_ID_RSP_USB_SPI_CONFIG.to_le_bytes(),
+                Error::CommunicationError("Unrecognized reponse to GET_USB_SPI_CONFIG"));
+        let max_write_chunk = u16::from_le_bytes(buf[2..4].try_into()?);
+        let max_read_chunk = u16::from_le_bytes(buf[4..6].try_into()?);
+        let feature_bitmap = u16::from_le_bytes(buf[6..8].try_into()?);
+        // Verify that interface supports concurrent read/write.
+        ensure!((feature_bitmap & 0x0001) != 0,
+                Error::CommunicationError("HyperDebug does not support bidirectional SPI"));
+        
+        Ok(Self {
+            inner: Rc::clone(&hyperdebug.inner),
+            interface: hyperdebug.spi_interface,
+            _target_idx: idx,
+            max_chunk_size: std::cmp::min(max_write_chunk, max_read_chunk) as usize,
+        })
+    }
+
+    /// Transmit data for a single SPI operation, using one or more USB packets.
+    fn transmit(&self, wbuf: &[u8], rbuf_len: usize) -> Result<()> {
+        let mut buf: [u8; USB_MAX_SIZE] = [0; USB_MAX_SIZE];
+        buf[0..2].clone_from_slice(&USB_SPI_PKT_ID_CMD_TRANSFER_START.to_le_bytes());
+        buf[2..4].clone_from_slice(&(wbuf.len() as u16).to_le_bytes());
+        buf[4..6].clone_from_slice(&(rbuf_len as u16).to_le_bytes());
+        let databytes = std::cmp::min(USB_MAX_SIZE - 6, wbuf.len());
+        buf[6..6 + databytes].clone_from_slice(&wbuf[0..databytes]);
+        self.usb_write_bulk(&buf[0..6 + databytes])?;
+        let mut index = databytes;
+
+        while index < wbuf.len() {
+            buf[0..2].clone_from_slice(&USB_SPI_PKT_ID_CMD_TRANSFER_CONTINUE.to_le_bytes());
+            buf[2..4].clone_from_slice(&(index as u16).to_le_bytes());
+            let databytes = std::cmp::min(USB_MAX_SIZE - 4, wbuf.len() - index);
+            buf[4..4 + databytes].clone_from_slice(&wbuf[index..index+databytes]);
+            self.usb_write_bulk(&buf[0..4 + databytes])?;
+            index += databytes;
+        }
+        Ok(())
+    }
+    
+    /// Receive data for a single SPI operation, using one or more USB packets.
+    fn receive(&self, rbuf: &mut [u8]) -> Result<()> {
+        let mut buf: [u8; USB_MAX_SIZE] = [0; USB_MAX_SIZE];
```

Comment:
You shouldn't have to repeat the type; write `0u8` in the RHS

Reply:
Done, here and elsewhere in this file.